indico/modules/categories/serialize.py

```diff
@@ -55,7 +58,12 @@ def serialize_category_ical(category, user, event_filter):
                       own_room_strategy,
                       own_venue_strategy)
              .order_by(Event.start_dt))
-    events = [e for e in query if e.can_access(user)]
+    if update_query:
+        query = update_query(query)
+    events = query.all()
```

Comment:
In my understanding, by doing it the way I said the list would only be built once. `.all()` builds it once and then the list comprehension below does it over again. Not to mention that if there's an `event_filter_fn` you get an extra list coming form `filter` (as opposed to `ifilter` that would defer the check).
As for the naming, couldn't we just use `query` until we finally do `events = [...]`?
Anyway, the performance penalty is probably zero, not important, just wondering.